descartes_moveit/include/descartes_moveit/moveit_state_adapter.h

```diff
@@ -85,7 +86,16 @@ class MoveitStateAdapter : public descartes_core::RobotModel
    * @brief Pointer to moveit robot state (mutable object state is reset with
    * each function call
    */
+
+  /**
+   * TODO: Checks for collisions at this joint pose. The setCollisionCheck(true) must have been
```

Comment:
This is a protected method so it is not meant to be called directly by the implementer.  I placed the collision code inside this method so that it can be conveniently invoked when needed by any of the interface methods.  There is already an **isValid** method which should also check for collisions when these are enabled; adding this method to the interface might make matters more confusing in my opinion.